project/chapters/templates/chapters/chapter.html

```diff
@@ -8,10 +8,15 @@
 <h1>{{ page.title }}</h1>
 
 {{ page.introduction | richtext }}
-{% if member == True %}
-    <a class="btn btn-secondary" href="{% pageurl page %}?leave=true">Leave Chapter</a><br><br>
-{% else %}
-    <a class="btn btn-primary" href="{% pageurl page %}?join=true">Join Chapter</a><br><br>
-{% endif %}
+<form action="." method="post">
+    {% csrf_token %}
+    {% if member == False %}
+        <input id="membership_action" type="hidden" name="membership_action" value="join">
+        <input type="submit" class="btn btn-primary"value="Join group">
```

Comment:
Let's add i18n support here
```suggestion
        <input type="submit" class="btn btn-primary"value="{% translate 'Join group' %}">
```